frontend/src/api/tests/DeviceApi.test.js

```diff
@@ -30,23 +30,18 @@ describe("DeviceApi", () => {
           () => new Date(`${sampleSendersResponse[0].lastCommunication}Z`)
         );
 
-      DeviceApi.getSenders((result) => {
-        try {
-          expect(axios.get).toHaveBeenCalledWith(
-            "http://localhost:8080/encoder",
-            authorizationHeader
-          );
-          expect(result).toEqual(DeviceFixture.getExpectedSendersResponse());
-          done();
-        } catch (error) {
-          done(error);
-        }
-      });
+      const result = await DeviceApi.getSenders();
+      expect(axios.get).toHaveBeenCalledWith(
+        "http://localhost:8080/encoder",
+        authorizationHeader
+      );
+
+      expect(result).toEqual(DeviceFixture.getExpectedSendersResponse());
```

Comment:
could benefit from being `toStrictEqual` but won't actually change test passing, will leave this one up to you again and refrain from further comments of this kind on the pr!

Reply:
d3ce970